project-generation/helpers.go

```diff
@@ -56,52 +101,86 @@ func ValidateAppName(ctx context.Context, unvalidatedAppName string) (validatedA
-	typeInputValid := false
-	validTypes := []string{
-		"generic-project", "base-application", "api", "controller", "event-driven",
-	}
-	for !typeInputValid {
-		if !stringInSlice(unvalidatedType, validTypes) {
-			typeInputValid = false
-			unvalidatedType, err = promptForInput(ctx, "Please specify the project type. This can have one of the "+
-				"following values: 'generic-project', 'base-application', 'api', 'controller', 'event-driven'")
-			if err != nil {
-				return validatedProjectType, err
-			}
-		} else {
-			typeInputValid = true
-			validatedProjectType = unvalidatedType
+func ValidateProjectType(ctx context.Context, projectType string) (validatedProjectType string, err error) {
+	if projectType == "" || projectType == "unset" {
+		prompt := "Please specify the project type"
+		options := []string{"generic-project", "base-application", "api", "controller", "event-driven"}
+		projectType, err = OptionPromptInput(ctx, prompt, options...)
+		if err != nil {
+			return "", err
+		}
+	}
+	return projectType, err
+}
+
+func ValidateGoVersion(ctx context.Context, unvalidatedGoVersion string) (validatedGoVersion string, err error) {
+	if unvalidatedGoVersion == "unset" || unvalidatedGoVersion == "" {
+		validatedGoVersion, err = PromptForInput(ctx, "Please specify the version of GO to use")
+		if err != nil {
+			return "", err
 		}
+	} else {
+		validatedGoVersion = unvalidatedGoVersion
 	}
-	return validatedProjectType, err
+	return validatedGoVersion, err
 }
 
 // ValidateProjectLocation will ensure that the projects location has been provided and is acceptable.
 // It will ensure the directory exists and has the option to offer a purge of files at that location
-func ValidateProjectLocation(ctx context.Context, unvalidatedProjectLocation, appName string, offerPurge bool) (validatedProjectLocation string, err error) {
-	if unvalidatedProjectLocation == "unset" {
-		validatedProjectLocation, err = promptForInput(ctx, "Please specify a directory for the project to be created in")
+func ValidateProjectLocation(ctx context.Context, unvalidatedProjectLocation string) (validatedProjectLocation string, err error) {
+	if unvalidatedProjectLocation == "unset" || unvalidatedProjectLocation == "" {
+		validatedProjectLocation, err = PromptForInput(ctx, "Please specify a directory for the project to be created in")
 		if err != nil {
-			return validatedProjectLocation, err
+			return "", err
 		}
 	} else {
 		validatedProjectLocation = unvalidatedProjectLocation
 	}
 	if validatedProjectLocation[len(validatedProjectLocation)-1:] != "/" {
 		validatedProjectLocation = validatedProjectLocation + "/"
 	}
-	if offerPurge {
-		err = OfferPurgeProjDestination(ctx, validatedProjectLocation, appName)
+	fmt.Println("validatedProjectLocation" + validatedProjectLocation)
+	return validatedProjectLocation, err
+}
+
+func ValidateProjectDirectory(ctx context.Context, path, projectName string) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
```

Comment:
If this case is true, return after log. Then the rest of the function doesn't need to be in an else case

Reply:
This may be true a few times in this function